Description: make scilab work with java >= 9
Author: Emmanuel Bourg & Julien Puydt
Forwarded: http://bugzilla.scilab.org/show_bug.cgi?id=11182

--- scilab.orig/scilab/modules/xcos/src/java/org/scilab/modules/xcos/BrowserView.java
+++ scilab/scilab/modules/xcos/src/java/org/scilab/modules/xcos/BrowserView.java
@@ -257,7 +257,7 @@
         final DefaultMutableTreeNode root = (DefaultMutableTreeNode) model.getRoot();
 
         DefaultMutableTreeNode node;
-        for (Enumeration<DefaultMutableTreeNode> e = root.depthFirstEnumeration(); e.hasMoreElements();) {
+        for (Enumeration<DefaultMutableTreeNode> e = (Enumeration) root.depthFirstEnumeration(); e.hasMoreElements();) {
             node = e.nextElement();
             BrowserTreeNodeData data = (BrowserTreeNodeData) node.getUserObject();
             if (data.getId() == uid) {
--- scilab.orig/scilab/modules/xcos/src/java/org/scilab/modules/xcos/block/actions/SuperblockMaskCustomizeAction.java
+++ scilab/scilab/modules/xcos/src/java/org/scilab/modules/xcos/block/actions/SuperblockMaskCustomizeAction.java
@@ -759,7 +759,7 @@
                  * it
                  */
                 @SuppressWarnings("unchecked")
-                final List < List<? >> data = model.getDataVector();
+                final List < List<? >> data = (List) model.getDataVector();
 
                 /* Inverting data */
                 List<?> tmp = data.get(row2);
--- scilab.orig/scilab/modules/xcos/src/java/org/scilab/modules/xcos/palette/model/Palette.java
+++ scilab/scilab/modules/xcos/src/java/org/scilab/modules/xcos/palette/model/Palette.java
@@ -58,7 +58,7 @@
      * @see javax.swing.tree.TreeNode#children()
      */
     @Override
-    public Enumeration<Object> children() {
+    public Enumeration<? extends TreeNode> children() {
         return null;
     }
 
--- scilab.orig/scilab/bin/scilab
+++ scilab/scilab/bin/scilab
@@ -477,7 +477,7 @@
 # Browse "default directory" (if it means anything under Unix/Linux) of Java
 # to find where it could be !
             if test -z "$JAVA_HOME"; then
-                JAVA_DEFAULT_DIR="/usr/java/jdk1.8.*/ /usr/lib/jvm/java-1.8.0-*/ /usr/lib/jvm/java-8-*/ /etc/alternatives/java_sdk /etc/alternatives/jre /usr/java/jdk*/ /usr/lib/jvm/java-*/"
+                JAVA_DEFAULT_DIR="/usr/lib/jvm/default-java /usr/java/jdk1.8.*/ /usr/lib/jvm/java-1.8.0-*/ /usr/lib/jvm/java-8-*/ /etc/alternatives/java_sdk /etc/alternatives/jre /usr/java/jdk*/ /usr/lib/jvm/java-*/"
                 for DIR in $JAVA_DEFAULT_DIR ; do
                     if test $SCIVERBOSE -ne 0; then
                         echo "Trying to find Java in $DIR "
@@ -639,7 +639,7 @@
     fi # IS_SCILAB_BINARY
 
     # Check if the lib exists
-    LIBJAVA="$JRE_HOME/lib/$proc/$JAVA_SHLIB"
+    LIBJAVA="$JRE_HOME/lib/$JAVA_SHLIB"
     if test ! -f "$LIBJAVA"; then
         if test $SCIVERBOSE -ne 0; then
             echo "Cannot find $LIBJAVA"
@@ -648,7 +648,7 @@
         CLASSPATH="${CLASSPATH:+:$CLASSPATH}$SCI/modules/core/jar/org.scilab.modules.core.jar"
         # @TODO This is the default path of the official sun JVM.
         # It is different for other implementation of the JVM....
-        LD_LIBRARY_PATH="$JRE_HOME/lib/$proc/:$JRE_HOME/lib/$proc/server/:$JRE_HOME/lib/$proc/native_threads/${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
+        LD_LIBRARY_PATH="$JRE_HOME/lib:$JRE_HOME/lib/$proc/:$JRE_HOME/lib/server/:$JRE_HOME/lib/$proc/server/:$JRE_HOME/lib/$proc/native_threads/${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
     fi
 
     # Bug http://bugzilla.scilab.org/show_bug.cgi?id=14290 -
