Description: Simplify command-line for the helper
Author: Sylvestre Ledru
Forwarded: Debian-specific

--- scilab.orig/scilab/modules/helptools/src/java/org/scilab/modules/helptools/JarOnlyConverter.java
+++ scilab/scilab/modules/helptools/src/java/org/scilab/modules/helptools/JarOnlyConverter.java
@@ -78,7 +78,7 @@
             directory.mkdirs();
             String jhs = new File(outputJavaHelp).getAbsolutePath();
 
-            String[] args = new String[] {"-db", jhs, "-nostopwords", "."};
+            String[] args = new String[] {"-nostopwords", "."};
 
             indexer.compile(args);
         } catch (Exception e) {
