Author: Julien Puydt <jpuydt@debian.org>
Description: make it build with Lucene 4.10
Forwarded: http://bugzilla.scilab.org/show_bug.cgi?id=15451

--- scilab.orig/scilab/modules/xcos/src/java/org/scilab/modules/xcos/palette/PaletteSearchManager.java
+++ scilab/scilab/modules/xcos/src/java/org/scilab/modules/xcos/palette/PaletteSearchManager.java
@@ -37,6 +37,7 @@
 import org.apache.lucene.index.IndexWriterConfig.OpenMode;
 import org.apache.lucene.store.Directory;
 import org.apache.lucene.store.RAMDirectory;
+import org.apache.lucene.util.Version;
 import org.scilab.modules.xcos.palette.model.PaletteBlock;
 import org.scilab.modules.xcos.palette.model.PaletteNode;
 import org.scilab.modules.xcos.palette.model.PreLoaded;
@@ -66,7 +67,7 @@
         try {
             indexedDirectory = new RAMDirectory();
             analyzer = new StandardAnalyzer();
-            IndexWriterConfig config = new IndexWriterConfig(analyzer);
+            IndexWriterConfig config = new IndexWriterConfig(Version.LATEST, analyzer);
             config.setOpenMode(OpenMode.CREATE);
             writer = new IndexWriter(indexedDirectory, config);
         } catch (IOException ex) {
