Description: fix compilation issue with new libicu (Closes: #962555)
Author: Boyuan Yang?
Forwarded: comes from ubuntu

--- a/scilab/modules/functions_manager/src/cpp/dynamic_modules.cpp
+++ b/scilab/modules/functions_manager/src/cpp/dynamic_modules.cpp
@@ -17,15 +17,15 @@
 #include "context.hxx"
 #include "configvariable.hxx"
 
+    //XML API
+#include <libxml/xpath.h>
+#include <libxml/xmlreader.h>
 
 extern "C"
 {
 #include "loadOnUseClassPath.h"
 #include "Scierror.h"
 }
-    //XML API
-#include <libxml/xpath.h>
-#include <libxml/xmlreader.h>
 
 extern "C"
 {
--- a/scilab/modules/io/src/cpp/loadlib.cpp
+++ b/scilab/modules/io/src/cpp/loadlib.cpp
@@ -17,6 +17,8 @@
 #include "context.hxx"
 #include "loadlib.hxx"
 #include "macrofile.hxx"
+#include <libxml/xpath.h>
+#include <libxml/xmlreader.h>
 extern "C"
 {
 #include "FileExist.h"
@@ -27,8 +29,6 @@
 #include "PATH_MAX.h"
 #include "pathconvert.h"
 }
-#include <libxml/xpath.h>
-#include <libxml/xmlreader.h>
 
 #define DEFAULT_ENCODING "UTF-8"
 
--- a/scilab/modules/scicos/src/cpp/XMIResource.cpp
+++ b/scilab/modules/scicos/src/cpp/XMIResource.cpp
@@ -12,9 +12,7 @@
 
 #include "XMIResource.hxx"
 
-extern "C" {
 #include <libxml/xmlversion.h>
-}
 
 namespace org_scilab_modules_scicos
 {
--- a/scilab/modules/scicos/src/cpp/XMIResource_load.cpp
+++ b/scilab/modules/scicos/src/cpp/XMIResource_load.cpp
@@ -21,13 +21,12 @@
 #include <cstdlib> // for atoi and atof
 #include <cstring> // for strcmp and strchr
 
-extern "C"
-{
 #include <libxml/tree.h>
 #include <libxml/parser.h>
 #include <libxml/xmlerror.h>
 #include <libxml/xmlreader.h>
-
+extern "C"
+{
 #include "sci_types.h"
 #include "sciprint.h"
 }
--- a/scilab/modules/scicos/src/cpp/XMIResource_save.cpp
+++ b/scilab/modules/scicos/src/cpp/XMIResource_save.cpp
@@ -19,10 +19,9 @@
 #include <vector>
 #include <cmath> // for trunc
 
+#include <libxml/xmlwriter.h>
 extern "C" {
 #include "sci_types.h"
-
-#include <libxml/xmlwriter.h>
 }
 
 namespace org_scilab_modules_scicos
--- a/scilab/modules/xml/sci_gateway/cpp/sci_htmlWrite.cpp
+++ b/scilab/modules/xml/sci_gateway/cpp/sci_htmlWrite.cpp
@@ -17,6 +17,7 @@
 #include "XMLDocument.hxx"
 #include "XMLElement.hxx"
 
+#include "libxml/tree.h"
 extern "C"
 {
 #include <string.h>
@@ -24,7 +25,6 @@
 #include "Scierror.h"
 #include "api_scilab.h"
 #include "xml_mlist.h"
-#include "libxml/tree.h"
 #include "expandPathVariable.h"
 #include "sci_malloc.h"
 #include "localization.h"
--- a/scilab/modules/xml/sci_gateway/cpp/sci_xmlAppend.cpp
+++ b/scilab/modules/xml/sci_gateway/cpp/sci_xmlAppend.cpp
@@ -17,13 +17,13 @@
 #include "XMLDocument.hxx"
 #include "XMLElement.hxx"
 
+#include "libxml/tree.h"
 extern "C"
 {
 #include "gw_xml.h"
 #include "Scierror.h"
 #include "api_scilab.h"
 #include "xml_mlist.h"
-#include "libxml/tree.h"
 #include "localization.h"
 }
 
--- a/scilab/modules/xml/sci_gateway/cpp/sci_xmlElement.cpp
+++ b/scilab/modules/xml/sci_gateway/cpp/sci_xmlElement.cpp
@@ -17,13 +17,13 @@
 #include "XMLDocument.hxx"
 #include "XMLElement.hxx"
 
+#include "libxml/tree.h"
 extern "C"
 {
 #include "gw_xml.h"
 #include "Scierror.h"
 #include "api_scilab.h"
 #include "xml_mlist.h"
-#include "libxml/tree.h"
 #include "localization.h"
 }
 
--- a/scilab/modules/xml/sci_gateway/cpp/sci_xmlWrite.cpp
+++ b/scilab/modules/xml/sci_gateway/cpp/sci_xmlWrite.cpp
@@ -17,6 +17,7 @@
 #include "XMLDocument.hxx"
 #include "XMLElement.hxx"
 
+#include "libxml/tree.h"
 extern "C"
 {
 #include <string.h>
@@ -24,7 +25,6 @@
 #include "Scierror.h"
 #include "api_scilab.h"
 #include "xml_mlist.h"
-#include "libxml/tree.h"
 #include "expandPathVariable.h"
 #include "sci_malloc.h"
 #include "localization.h"
--- a/scilab/modules/xml/src/cpp/XMLDocument.hxx
+++ b/scilab/modules/xml/src/cpp/XMLDocument.hxx
@@ -24,12 +24,9 @@
 #include "dynlib_xml_scilab.h"
 
 #include "xml.h"
-extern "C"
-{
 #ifndef XML_XPATH_CHECKNS
 #define XML_XPATH_CHECKNS
 #endif
-}
 
 #include "XMLObject.hxx"
 
