Description: more recent libxml2 doesn't ship xml2-config anymore
Author: Julien Puydt
Forwarded: http://bugzilla.scilab.org/show_bug.cgi?id=16352

--- a/scilab/configure.ac
+++ b/scilab/configure.ac
@@ -1620,7 +1620,9 @@
 ## Test for libxml
 #######################
 
-AC_LIBXML2()
+PKG_CHECK_MODULES([XML], [libxml-2.0])
+AC_SUBST(XML_CFLAGS)
+AC_SUBST(XML_LIBS)
 
 #######################
 ## Test for gettext
@@ -2503,9 +2505,8 @@
 
 echo ""
 echo "XML configuration:"
-echo "  XML_FLAGS .......... = $XML_FLAGS"
+echo "  XML_CFLAGS .......... = $XML_CFLAGS"
 echo "  XML_LIBS ........... = $XML_LIBS"
-echo "  XML_VERSION ........ = $XML_VERSION"
 
 echo ""
 echo "HDF5 configuration:"
--- a/scilab/modules/completion/Makefile.am
+++ b/scilab/modules/completion/Makefile.am
@@ -76,7 +76,7 @@
     -I$(top_srcdir)/modules/dynamic_link/includes \
     -I$(top_srcdir)/modules/external_objects/includes \
     $(JAVA_JNI_INCLUDE) \
-    $(XML_FLAGS) \
+    $(XML_CFLAGS) \
     $(AM_CPPFLAGS)
 
 # Name of the future library (.la will be replace by .so, .a, etc regarding
--- a/scilab/modules/core/Makefile.am
+++ b/scilab/modules/core/Makefile.am
@@ -190,7 +190,7 @@
     -I$(top_srcdir)/modules/fftw/includes/ \
     -I$(top_srcdir)/modules/ui_data/includes/ \
     $(HDF5_CFLAGS) \
-    $(XML_FLAGS) \
+    $(XML_CFLAGS) \
     $(AM_CPPFLAGS)
 
 # Used by sci_getdebuginfo:
--- a/scilab/modules/coverage/Makefile.am
+++ b/scilab/modules/coverage/Makefile.am
@@ -66,7 +66,7 @@
     -I$(top_srcdir)/modules/string/includes/ \
     -I$(top_srcdir)/modules/dynamic_link/includes \
     -I$(top_srcdir)/modules/threads/includes/ \
-    $(XML_FLAGS) \
+    $(XML_CFLAGS) \
     $(COVERAGE_FLAGS) \
     $(AM_CPPFLAGS)
 
--- a/scilab/modules/fileio/Makefile.am
+++ b/scilab/modules/fileio/Makefile.am
@@ -134,7 +134,7 @@
 	-I$(top_srcdir)/modules/dynamic_link/includes \
 	-I$(top_srcdir)/modules/preferences/includes \
 	$(HDF5_CFLAGS) \
-	$(XML_FLAGS) \
+	$(XML_CFLAGS) \
 	$(LIBARCHIVE_CFLAGS) \
 	$(AM_CPPFLAGS)
 
--- a/scilab/modules/functions_manager/Makefile.am
+++ b/scilab/modules/functions_manager/Makefile.am
@@ -45,7 +45,7 @@
 	-I$(top_srcdir)/modules/string/includes/ \
 	-I$(top_srcdir)/modules/threads/includes/ \
 	$(AM_CPPFLAGS) \
-	$(XML_FLAGS)
+	$(XML_CFLAGS)
 
 libscifunctions_manager_la_includedir=$(pkgincludedir)
 
--- a/scilab/modules/io/Makefile.am
+++ b/scilab/modules/io/Makefile.am
@@ -56,7 +56,7 @@
     -I$(top_srcdir)/modules/console/includes \
     -I$(top_srcdir)/modules/hdf5/includes \
     -I$(top_srcdir)/modules/tclsci/includes \
-    $(XML_FLAGS) \
+    $(XML_CFLAGS) \
     $(AM_CPPFLAGS)
 
 if MAINTAINER_MODE
--- a/scilab/modules/jvm/Makefile.am
+++ b/scilab/modules/jvm/Makefile.am
@@ -50,7 +50,7 @@
     -I$(top_srcdir)/modules/fileio/includes/ \
     -I$(top_srcdir)/modules/dynamic_link/includes \
     -I$(top_srcdir)/modules/preferences/includes/ \
-    $(XML_FLAGS) \
+    $(XML_CFLAGS) \
     $(AM_CPPFLAGS)
 
 pkglib_LTLIBRARIES = libscijvm-disable.la
--- a/scilab/modules/preferences/Makefile.am
+++ b/scilab/modules/preferences/Makefile.am
@@ -64,7 +64,7 @@
     -I$(srcdir)/includes/ \
     -I$(srcdir)/src/c/ \
     -I$(srcdir)/src/jni/ \
-    $(XML_FLAGS) \
+    $(XML_CFLAGS) \
     -I$(top_srcdir)/modules/ast/includes/ast/ \
     -I$(top_srcdir)/modules/ast/includes/exps/ \
     -I$(top_srcdir)/modules/ast/includes/operations/ \
--- a/scilab/modules/scicos/Makefile.am
+++ b/scilab/modules/scicos/Makefile.am
@@ -155,13 +155,13 @@
 			-I$(top_srcdir)/modules/threads/includes/ \
 			-I$(top_srcdir)/modules/console/includes/ \
 			-I$(top_srcdir)/modules/differential_equations/includes/ \
-			$(XML_FLAGS) \
+			$(XML_CFLAGS) \
 			$(AM_CPPFLAGS)
 else
 libsciscicos_algo_la_CPPFLAGS= -I$(top_srcdir)/modules/output_stream/includes/ \
 			-I$(top_srcdir)/modules/api_scilab/includes/ \
 			-I$(top_srcdir)/modules/scicos/includes/ \
-			$(XML_FLAGS) \
+			$(XML_CFLAGS) \
 			$(AM_CPPFLAGS)
 endif
 libsciscicos_cli_la_CPPFLAGS = $(libsciscicos_algo_la_CPPFLAGS)
--- a/scilab/modules/slint/Makefile.am
+++ b/scilab/modules/slint/Makefile.am
@@ -126,7 +126,7 @@
     -I$(top_srcdir)/modules/string/includes/ \
     -I$(top_srcdir)/modules/dynamic_link/includes \
     -I$(top_srcdir)/modules/threads/includes/ \
-    $(XML_FLAGS) \
+    $(XML_CFLAGS) \
     $(PCRE_CFLAGS) \
     $(SLINT_FLAGS) \
     $(AM_CPPFLAGS)
--- a/scilab/modules/xcos/Makefile.am
+++ b/scilab/modules/xcos/Makefile.am
@@ -119,7 +119,7 @@
     -I$(top_srcdir)/modules/localization/includes/ \
     -I$(top_srcdir)/modules/fileio/includes/ \
     -I$(top_srcdir)/modules/api_scilab/includes/ \
-    $(XML_FLAGS) \
+    $(XML_CFLAGS) \
     $(JAVA_JNI_INCLUDE) \
     $(AM_CPPFLAGS)
 
@@ -135,7 +135,7 @@
     -I$(top_srcdir)/modules/ast/includes/types/ \
     -I$(top_srcdir)/modules/localization/includes/ \
     -I$(top_srcdir)/modules/output_stream/includes \
-    $(XML_FLAGS) \
+    $(XML_CFLAGS) \
     $(AM_CPPFLAGS)
 
 XCOS_DISABLE_C_SOURCES = src/noxcos/noxcos.c
--- a/scilab/modules/xml/Makefile.am
+++ b/scilab/modules/xml/Makefile.am
@@ -109,7 +109,7 @@
     -I$(top_srcdir)/modules/fileio/includes/ \
     -I$(top_srcdir)/modules/string/includes/ \
     -I$(top_srcdir)/modules/dynamic_link/includes \
-    $(XML_FLAGS) \
+    $(XML_CFLAGS) \
     $(AM_CPPFLAGS)
 
 pkglib_LTLIBRARIES = libscixml.la
--- a/scilab/modules/webtools/Makefile.am
+++ b/scilab/modules/webtools/Makefile.am
@@ -50,7 +50,7 @@
     -I$(top_srcdir)/modules/api_scilab/includes \
     -I$(top_srcdir)/modules/preferences/includes \
     $(CURL_CFLAGS) \
-    $(XML_FLAGS) \
+    $(XML_CFLAGS) \
     $(AM_CPPFLAGS)
 
 libsciwebtools_algo_la_CPPFLAGS = $(libsciwebtools_la_CPPFLAGS)
